src/main/java/io/jenkins/plugins/gitlabserver/servers/GitLabServers.java

```diff
@@ -107,18 +112,19 @@ public String getDisplayName() {
      *
      * @param endpoints the list of endpoints.
      */
-    public void setServers(@CheckForNull List<? extends GitLabServer> endpoints) {
+    public synchronized void setServers(@CheckForNull List<? extends GitLabServer> endpoints) {
```

Comment:
Do these methods *really* require synchronization?

Reply:
I am not quite sure about that. This is one of the few parts that I do not understand clearly but made a leap of faith as `Gitea Plugin` has done expecting there would be a use case for this in branch source. Let me know what is your suggestion.

Reply:
Is this only called from the config UI? In that case, the only race condition I can think of would be two admins making changes to the config at the same time. I suppose that's technically possible, thought unlikely

Reply:
Thanks @jeffpearce for pointing it out. I think it is a good idea to keep it synchronized.

Reply:
I believe there is synchronization further below this in serialization. Generally, I tend to default to not synchronizing if there isn't a very good reason for it because it can cause performance bottlenecks or, worse, deadlocks (not saying that you have the deadlock case here).

Reply:
This is a trade-off then! Now, I think maybe let's not keep methods `synchronized`. If there is an essential requirement that comes up, we may re-implement it. Let me know your thoughts.